pkg/types/validation/installconfig.go

```diff
@@ -224,6 +224,12 @@ func validateNetworkingIPVersion(n *types.Networking, p *types.Platform) field.E
 		case p.Azure != nil && experimentalDualStackEnabled:
 			logrus.Warnf("Using experimental Azure dual-stack support")
 		case p.BareMetal != nil:
+			// VIP for API should belong to IPv4 family in Bare Metal dual-stack
+			// Because primary network should also be IPv4.
+			apiVIP := net.ParseIP(p.BareMetal.APIVIP)
+			if apiVIP.To4() == nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("networking", "baremetal", "apiVIP"), p.BareMetal.APIVIP, "VIP for the API must be in IPv4 network for dual-stack IPv4/IPv6"))
```

Comment:
The fact that "primary network should be IPv4" is a *bug*. We are supposed to support either IPv4-primary or IPv6-primary, and eventually we will, and we should minimize the number of places that we need to change things in the installer when that happens.

The constraint on the API VIP should be "the API VIP must be of the cluster's primary IP family, whatever family that is",